core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -3243,13 +3250,13 @@ class ViewMapping<
     const size_t alloc_size =
         (m_impl_offset.span() * MemorySpanSize + MemorySpanMask) &
         ~size_t(MemorySpanMask);
-
+    const std::string& alloc_name =
+        ((Kokkos::Impl::ViewCtorProp<void, std::string> const&)arg_prop).value;
```

Comment:
Same comment about C-style casts

Reply:
How true to the old code do we want to be (I'm ripping from old code here). I'd like to make it a `static_cast`, but it started as a C-cast, which is more of a `reinterpret_cast`. If I make the change, should I change the other C-casts in the function?

Reply:
I would prefer a `reinterpret_cast`.  You can change all occurrence of `(Kokkos::Impl::ViewCtorProp<void, std::string> const&)arg_prop` if you agree that this is an improvement.

Reply:
It's not a `reinterpret_cast`, it's a `static_cast`. C spells both of those the same way. This is being used to disambiguate a base class; i.e., I believe the following would also work (though I doubt it would be much clearer for most users), but mostly see my comment above for what I think is a better solution

```c++
arg_prop.template ViewCtorProp<void, std::string>::value
```

Reply:
Yeah, I've verified that the move to `static_cast` fixes my bugs